gems/smithy/lib/smithy/anvil/client/views/shapes.rb

```diff
@@ -0,0 +1,241 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Anvil
+    module Client
+      module Views
+        # @api private
+        class Shapes < View
+          def initialize(plan)
+            @plan = plan
+            @model = plan.model
+            @service_shape = assemble_service_shape
+            @shapes, @operation_shapes = assemble_shapes
+            super()
+          end
+
+          def namespace
+            Tools::Namespace.namespace_from_gem_name(@plan.options[:gem_name])
+          end
+
+          def shapes_with_members
+            @shapes.reject { |s| s.members.empty? }
+          end
+
+          private
+
+          def assemble_service_shape
```

Comment:
Naming nit, build_ is a fine prefix. But also, this could simply just be a method called service_shape? There is no need to make an instance variable IMO. Possibly the same for other cases.

Reply:
I'd lean towards a build_ prefix for these.